analysis/processing/data_process_baseline.R

```diff
@@ -0,0 +1,174 @@
+
+
+# Load functions
+source(here::here("analysis", "custom_functions.R"))
+
+end_date = as.Date("2023-02-04")
+
+
+#######################################
+# Prepare data
+#######################################
+
+# Read in and clean data 
+baseline <- read_feather(here::here("output", "input_baseline.feather")) %>%
+  mutate_at(c(vars(c(contains("_date")))), as.Date, format = "%Y-%m-%d") %>%    
+  mutate(dob = as.Date(as.character(as.POSIXct(dob)), format = "%Y-%m-%d"),
+         
+    # Set DOB to mid-month
+    dob = dob + 14,
+    
+    # Flag for clinically vulnerable people
+    cv = immunosuppressed | chronic_kidney_disease | chronic_resp_disease | 
+      diabetes | chronic_liver_disease | chronic_neuro_disease |  asplenia |
+      chronic_heart_disease | sev_mental | sev_obesity | asthma,
+    
+    # Determine earliest recorded date of flu vaccination 
+    flu_vax_date = pmin(flu_vax_med_date, flu_vax_tpp_date, 
+                        flu_vax_clinical_date, na.rm = TRUE),
+    
+    # Received booster anytime in 2022/23
+    booster = if_else(
+      (covid_vax_3_date >= as.Date("2022-09-05") & covid_vax_3_date < end_date) |
+      (covid_vax_4_date >= as.Date("2022-09-05") & covid_vax_4_date < end_date),
+      1, 0, 0),
+    
+    # Booster date (if received)
+    boost_date = if_else(booster == 1,
+                         pmin(covid_vax_3_date, covid_vax_4_date, na.rm = TRUE),
+                         NA_Date_),
+    
+    # If received fourth dose prior to second booster campaign
+    covid_vax4_early = if_else(covid_vax_4_date < as.Date("2022-09-05"), 1, 0, missing = 0),
+    
+    # If received third dose prior to first booster campaign
+    covid_vax3_early = if_else(covid_vax_3_date < as.Date("2021-09-16"), 1, 0, missing = 0),
+    
+    # Received another COVID vaccine in 3 months prior to start of campaign
+    covid_vax_recent = if_else(( 
+                                (covid_vax_3_date >= as.Date("2022-07-15") &
+                                 covid_vax_3_date <= as.Date("2022-10-15")) |
+                                (covid_vax_2_date >= as.Date("2022-07-15") &
+                                 covid_vax_2_date <= as.Date("2022-10-15")) |
+                                (covid_vax_1_date >= as.Date("2022-07-15") &
+                                 covid_vax_1_date <= as.Date("2022-10-15"))
+                                ), 1, 0, 0),
+    
+    # Received 2nd dose at least 3 months prior to start of campaign
+    covid_vax2 = if_else(covid_vax_2_date < as.Date("2022-07-15"), 1, 0, 0),
+    
+    # Received 3rd dose at least 3 months prior to start of campaign
+    covid_vax3 = if_else(covid_vax_3_date < as.Date("2022-07-15"), 1, 0, 0),
+    
+    # Flag for people prioritised for vaccine (including evidence of having received
+       # COVID vaccine before becoming available to general population)
+    vax_priority = cv | housebound | carehome | hscworker | covid_vax4_early |
```

Comment:
I don't think housebound is on the priority list. In the vaccine studies it was just an exclusion criterion we used to improve comparability. Not saying we shouldn't exclude them here, but including them in the "priority" definition is strictly speaking not true